### PR TITLE
Add permission prompts for macOS Catalina

### PR DESCRIPTION
**Pull request by stanczakp**
This update adds permissions prompt for MacOS Catalina to allow active-win to get window title.

### CODE REVIEW

**Comment by lkobylski** [comment]
I think it should resolve problem with empty title field on macOS Catalina. @sindresorhus .

File: Sources/active-win/main.swift [hunk 1]
```diff
@@ -5,9 +5,21 @@ func toJson<T>(_ data: T) throws -> String {
...
 
+if !AXIsProcessTrustedWithOptions(["AXTrustedCheckOptionPrompt": true] as CFDictionary) {
+    print("BActive-Win requires permission for accessibility in System Preferences > Security & Privacy > Privacy > Accessibility.")
+    exit(0)
+}
+
 let frontmostAppPID = NSWorkspace.shared.frontmostApplication!.processIdentifier
 let windows = CGWindowListCopyWindowInfo([.optionOnScreenOnly, .excludeDesktopElements], kCGNullWindowID) as! [[String: Any]]
 
+let windowImage = CGWindowListCreateImage(.null, .optionIncludingWindow, windows[0][kCGWindowNumber as String] as! CGWindowID, [.boundsIgnoreFraming, .bestResolution])
```
Comment (sindresorhus) [inline Sources/active-win/main.swift]: It should be clearer from the naming here that this is to trigger the permission dialog.

File: Sources/active-win/main.swift [hunk 2]
```diff
@@ -5,9 +5,21 @@ func toJson<T>(_ data: T) throws -> String {
 	return String(data: json, encoding: .utf8)!
 }
 
+if !AXIsProcessTrustedWithOptions(["AXTrustedCheckOptionPrompt": true] as CFDictionary) {
+    print("BActive-Win requires permission for accessibility in System Preferences > Security & Privacy > Privacy > Accessibility.")
+    exit(0)
```
Comment (sindresorhus) [inline Sources/active-win/main.swift]:
```suggestion
    exit(1)
```

File: Sources/active-win/main.swift [hunk 3]
```diff
@@ -5,9 +5,21 @@ func toJson<T>(_ data: T) throws -> String {
 	return String(data: json, encoding: .utf8)!
 }
 
+if !AXIsProcessTrustedWithOptions(["AXTrustedCheckOptionPrompt": true] as CFDictionary) {
+    print("BActive-Win requires permission for accessibility in System Preferences > Security & Privacy > Privacy > Accessibility.")
```
Comment (sindresorhus) [inline Sources/active-win/main.swift]: Typo

File: Sources/active-win/main.swift [hunk 4]
```diff
@@ -5,19 +5,19 @@ func toJson<T>(_ data: T) throws -> String {
...
 let frontmostAppPID = NSWorkspace.shared.frontmostApplication!.processIdentifier
 let windows = CGWindowListCopyWindowInfo([.optionOnScreenOnly, .excludeDesktopElements], kCGNullWindowID) as! [[String: Any]]
 
-let windowImage = CGWindowListCreateImage(.null, .optionIncludingWindow, windows[0][kCGWindowNumber as String] as! CGWindowID, [.boundsIgnoreFraming, .bestResolution])
-
-if windowImage == nil {
-	print("Active-Win requires permission for screen recording in System Preferences > Security & Privacy > Privacy > Screen Recording.")
-	exit(0)
+// Show screen recording permissions screen if not enabled; required to get the complete window title
+if CGWindowListCreateImage(.null, .optionIncludingWindow, windows[0][kCGWindowNumber as String] as! CGWindowID, [.boundsIgnoreFraming, .bestResolution]) == nil {
```
Comment (sindresorhus) [inline Sources/active-win/main.swift]: > windows[0][kCGWindowNumber as String] as! CGWindowID

What if there are no windows?

Comment (pstanczak-dyna) [reply Sources/active-win/main.swift]: I hope this is the proper way to check if first array element exists in swift ;)

Comment (stanczakp) [reply Sources/active-win/main.swift]: I hope this is the proper way to check for the first array element in swift.

**Pull request merged**